### ISSUE TITLE
DLND-Week5 1v1 问题集

### ISSUE THREAD

**Issue by chenghong-lin-nu** [Hclmaster/blog#14]
#### 1. hidden layer为什么可以用tf.layers.dense呢？我一直以为它只能用来构建全连接层。
- 因为隐藏层有多种，全连接层是隐藏层的一种，tf.layers.dense可以用于构建全连接层。

#### 1. d_loss = d_loss_real + d_loss_fake为什么是这样的呢？
- discriminator的loss是由两部分组成，因为这里discriminator不仅要学习怎样判断真货，还得要学习怎样判断假货，所以这里就有了2个loss。这里只是说明了discriminator的loss是由real和fake两部分组成。

#### 2. 对于real_images_logits来说，为什么在计算loss的时候它的labels要是1呢？
>For the real image logits, we'll use d_logits_real which we got from the discriminator in the cell above. For the labels, we want them to be all ones, since these are all real images.

- 因为对于discriminator来说，它就像验钞机一样，所以它要尽可能的验准确这个是不是真的钱。 所以对于real的来说，就要尽可能让它靠近真实值，也就是1。

#### 3. 对于fake_image_logits来说，为什么计算loss的时候它的labels要是0呢？
- 因为它是假币嘛，所以验钞机要尽可能的实现识别假币的工作，所以也就分配给它0的概率咯。

#### 4. 对于generator的loss来说，为什么这里的labels是1呢？
- 因为generator是一个生产假币的机器啊，所以对于验钞机返回的结果它要让它尽可能的靠近1才行，所以它这里就以1（可以以假乱真的假币）为目标，生产新的假币，所以这里的labels就是1了！
- 也就是说，它们两个网络其实是在相互竞争。

#### 5. discriminator的流程是：真实的图片数据输入进去是为了让它学会哪些是真的，而不是去用真实数据集去和假的进行比较是吗？
- 对的，没错的。

#### 6. 为什么最后discriminator的logits的units为1呢？
![image](https://user-images.githubusercontent.com/16880879/39685011-4d1a0cea-51f2-11e8-9c30-e0a145fb3812.png)
- 是因为discriminator只需判别输入图片的真伪，本质上就是一个二元分类器，所以最后的输出units为1，这个units代表的是一维的。
